weatherApp/R/weather_app_functions.R

```diff
@@ -0,0 +1,371 @@
+  }
+
+  # Write down the url that retrieves the long/lat information from openstreetmap.org
+
+  my_url <- paste("https://nominatim.openstreetmap.org/search?q=", my_location_string,
+                  "&format=json&limit=1", sep="")
+
+  # call the Url, retrieve the data and check if it was successful.
+  my_raw_results <- httr::GET(my_url)
+
+  if (status_code(my_raw_results)[[1]] != 200) {
+    stop("Please enter a valid location!")
+  }
+
+  # Transform the JSON data into useable format.
+  my_content <- httr::content(my_raw_results, as = "text")
+
+  my_content_from_json <- jsonlite::fromJSON(my_content)
+
+  # Save the longitude latitude data in an object and return it.
+  geocode_data <- data.frame(3)
+  geocode_data$location <- location
+  geocode_data$latitude <- my_content_from_json$lat
+  geocode_data$longitude <- my_content_from_json$lon
+
+  return(geocode_data)
+
+}
+
+# Example for geocode()
+# geocode("Amsterdam, Netherlands")
+
+
+# The following function takes the location (in string format) and your apiKey (in string format) as input.
+# The function returns the weather data for a specific location as a list.
+# The weather data are provided by www.openweathermap.org
+
+#' @export
+get_weather <- function(location, apiKey) {
+
+  # first, we need to use the function geocode(location) to get the longitude and the latitude of the desired location.
+
+  latitude <- geocode(location)$latitude
+  longitude <- geocode(location)$longitude
+
+  # latitude <- tidygeocoder::geo_osm(location)$lat[1] # old version
+  # longitude <- tidygeocoder::geo_osm(location)$long[1] # old version
+
+  if (is.na(latitude) | is.na(longitude)) {
+
+    stop("Please type in a valid location")
+
+  }
+
+  # now, we access www.openweatherapp.org and retrieve the weather data.
+
+  my_url <- paste0("https://api.openweathermap.org/data/2.5/onecall?lat=", latitude, "&lon=", longitude,
+                  "&exclude=FALSE&appid=", apiKey)
+
+  # We retrieve the data using the url
+  my_raw_results <- httr::GET(my_url)
+
+  # We check whether we successfully retrieved data from the API.
+
+  if (status_code(my_raw_results)[[1]] != 200) {
+
+    stop("Something went wrong. You might have put in a wrong location or apiKey. Please check for spelling mistakes and try again.")
+
+  }
+
+  # And we transform the infromation we retrieved from JSON format into a usable format in R .
+
+  my_content <- httr::content(my_raw_results, as = "text")
+
+  my_content_from_json <- jsonlite::fromJSON(my_content)
+
+  return(my_content_from_json)
+
+}
+
+## Example for get_weather
+# get_weather("Amsterdam, Niederlande", Sys.getenv("MY_API"))
+
+
+
+# This function gets as input whether the user wants the current weather forecast, the hourly forecast,
+# or the daily forecast, the location of interest and the apiKey. It returns a list with the desired wheather information.
+
+#' @export
+get_your_forecast <- function(cur_hour_day = "current", location, apiKey) {
```

Comment:
I am minimally confused why you  are not calling the function get_forecast to stick with the theme?